internal/iam/repository_group.go

```diff
@@ -127,15 +130,15 @@ func (r *Repository) LookupGroup(ctx context.Context, withPublicId string, opt .
 		db.ExpBackoff{},
 		func(read db.Reader, w db.Writer) error {
 			if err := read.LookupByPublicId(ctx, &g); err != nil {
-				return fmt.Errorf("lookup group: failed %w for %s", err, withPublicId)
+				return errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("for group %s", withPublicId)))
```

Comment:
I'm not a fan of this pattern.. I think, you're losing traceability.

Reply:
it's repeated quite a bit in the PR... you're dropping context from the original errors.

Reply:
once we decide what to do (if anything) I'll leave it to you to chase down all the instances of this pattern.

Reply:
I agree that you won't be able to do a search for say `lookup group: failed` to find the line of code in repository_group that is failing,  Which might not be ideal, However, I am not sure we really want to keep all the redundant message text in all these errors.  You can still pretty easily find the `op` and then search upwards, also using the wrapped errors op you should be able to find the call that is returning in the error.

You might have noticed in other instances I actually dropped the `for group` completely in the `doTx` and just had it wrapped once after the `doTx` when the error is actually returned.  I made a commit now that brings this inline.

Reply:
IMO `errors.New` is really where you want the unique error message to be traceable, as this is the location where you would most likely `"fix"` the error issue.  When wrapping the errors the `op` should be sufficient to trace the path, sometimes you do want additional context in the message but not always...
In this example, we would most likely be addressing any issue within `LookupByPublicId` and at this point within `LookupGroup`